packages/core/src/messages/responses/create-cache.ts

```diff
@@ -1,39 +1,23 @@
 import {SdkError} from '../../errors';
-import {ResponseBase, ResponseError, ResponseSuccess} from './response-base';
+import {BaseResponseError, BaseResponseSuccess} from './response-base';
+import {CreateCacheResponse} from './enums';
+
+interface IResponse {
+  readonly type: CreateCacheResponse;
+}
 
 /**
- * Parent response type for a create cache request.  The
- * response object is resolved to a type-safe object of one of
- * the following subtypes:
- *
- * - {Success}
- * - {AlreadyExists}
- * - {Error}
- *
- * `instanceof` type guards can be used to operate on the appropriate subtype.
- * @example
- * For example:
- * ```
- * if (response instanceof CreateCache.Error) {
- *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
- *   // `CreateCache.Error` in this block, so you will have access to the properties
- *   // of the Error class; e.g. `response.errorCode()`.
- * }
- * ```
+ * Indicates a successful create cache request.
  */
-export abstract class Response extends ResponseBase {}
-
-class _Success extends Response {}
+export class Success extends BaseResponseSuccess implements IResponse {
+  readonly type: CreateCacheResponse.Success = CreateCacheResponse.Success;
+}
 
 /**
- * Indicates a Successful create cache request.
+ * Indicates that the cache already exists.
  */
-export class Success extends ResponseSuccess(_Success) {}
-
-class _Error extends Response {
-  constructor(protected _innerException: SdkError) {
-    super();
-  }
+export class AlreadyExists extends BaseResponseSuccess implements IResponse {
+  readonly type: CreateCacheResponse.Success = CreateCacheResponse.Success;
```

Comment:
should this be a separate type from `success`? This doesn't seem like a successful request?

Reply:
good call, I can change that to extend ResponseBase instead